core/src/androidMain/kotlin/com/ustadmobile/core/catalog/contenttype/VideoTypePluginAndroid.kt

```
@@ -7,162 +7,275 @@ import android.media.MediaFormat
+                            Napier.d(tag = VIDEO_ANDROID, message = "progress at value ${progress * 100}")
+                            contentJobItem.cjiItemProgress = ((progress * contentJobItem.cjiItemTotal) / progressSize).toLong()
+                            jobProgress.onProgress(contentJobItem)
+                        }
+
+                        override fun onCompleted(id: String, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
+                            Napier.d(tag = VIDEO_ANDROID, message = "completed transform")
+                            contentJobItem.cjiItemProgress = contentJobItem.cjiItemTotal / progressSize
+                            jobProgress.onProgress(contentJobItem)
+                            videoCompleted.complete(true)
+                        }
+
+                        override fun onCancelled(id: String, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
+                            Napier.d(tag = VIDEO_ANDROID, message = "cancelled transform")
+                            videoCompleted.complete(false)
+                        }
+
+                        override fun onError(id: String, cause: Throwable?, trackTransformationInfos: MutableList<TrackTransformationInfo>?) {
+                            videoCompleted.completeExceptionally(cause
+                                    ?: RuntimeException("error on video id: $id"))
+                        }
+
+                    }, TransformationOptions.Builder()
+                            .setGranularity(MediaTransformer.GRANULARITY_DEFAULT)
+                            .setVideoFilters(null)
+                            .build()
+                    )
+
+
+                    try {
+                        videoCompleted.await()
+                    } catch (e: Exception) {
+                        Napier.e(tag = VIDEO_ANDROID, throwable = e, message = e.message ?: "")
```

I think if there is an exception here, we need to stop. Why continue creating and uploading a container if the conversion has failed? It should throw a fatal exception (which the runner will catch)